torch/onnx/symbolic_helper.py

```diff
@@ -375,6 +375,21 @@ def _is_all_integral(scalars):
     step = g.op("Cast", step, to_i=scalar_type_to_onnx[type]) if step else None
     return type, end, start, step
 
+torch_float_dtype_to_onnx = {
+    torch.half: torch.onnx.TensorProtoDataType.FLOAT16,
+    torch.float: torch.onnx.TensorProtoDataType.FLOAT,
+    torch.double: torch.onnx.TensorProtoDataType.DOUBLE
+}
+
+def _unary_ufunc_helper(g, self, op_name):
+    if not _is_fp(self):
+        dtype = torch.get_default_dtype()
+        assert dtype in torch_float_dtype_to_onnx.keys()
+        onnx_scalar_type = torch_float_dtype_to_onnx[dtype]
+        self = g.op("Cast", self, to_i=onnx_scalar_type)
```

Comment:
I think I found a simpler way to generate this cast that doesn't require the table above. Try:

`self = g.op("Cast", input, to_i=scalar_type_to_pytorch_type.index(torch.get_default_dtype()))`

Reply:
Sure! Let me try if it works fine.

Reply:
I think you meant:

```
type = scalar_type_to_pytorch_type.index(torch.get_default_dtype())
self = g.op("Cast", self, to_i=scalar_type_to_onnx[type])
```

Please correct me if I'm wrong.

Reply:
This is solved in the recent commit: https://github.com/pytorch/pytorch/pull/33322/commits/29ae207f362efe4040b598355cfe1db1fadb82ad. Thanks!